platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/ConcurrentArray.java

```diff
@@ -392,4 +376,40 @@ boolean add(T element) {
             return true;
         }
     }
+
+    /**
+     * Utility class to provide Stream support on top of concurrent arrays.
+     *
+     * @param <T>
+     */
+    private static class ConcurrentArraySpliterator<T> extends AbstractSpliterator<T> {
+        private SubArray<T> arr;
+        private int arrIndex = 0;
+
+        ConcurrentArraySpliterator(final int size, final SubArray<T> head) {
+            super(size, Spliterator.SIZED | Spliterator.NONNULL | Spliterator.IMMUTABLE);
```

Comment:
Should include `Spliterator.CONCURRENT` here so this ConcurrentArray can be used with `parallelStream`